drivers/sensor/lsm6dsl/Kconfig

```diff
@@ -8,13 +8,30 @@
 #
 
 menuconfig LSM6DSL
-	bool "LSM6DSL I2C accelerometer and gyroscope Chip"
-	depends on I2C
+	bool "LSM6DSL I2C/SPI accelerometer and gyroscope Chip"
+	depends on SENSOR && (I2C || SPI)
 	default n
 	help
-	  Enable driver for LSM6DSL I2C-based accelerometer and gyroscope
+	  Enable driver for LSM6DSL accelerometer and gyroscope
 	  sensor.
 
+choice
+	prompt "Interface to AP"
+	depends on LSM6DSL
+	default LSM6DSL_SPI
+	help
+	  Select interface the LSM6DSL driver is connected to AP
+
+config LSM6DSL_SPI
+	depends on SPI
+	bool "SPI Interface"
+
+config LSM6DSL_I2C
+	depends on I2C
+	bool "I2C Interface"
+
+endchoice
+
 if !HAS_DTS_I2C_DEVICE
```

Comment:
We need to adapt here.
Actually driver could be compiled with this flag just because it is used for other drivers and then LSM6DSL_DEV_NAME is no more visible.

So LSM6DSL_DEV_NAME should be outside if !HAS_DTS_I2C_DEVICE
and should have following dependency:
```depends on (LSM6DSL_I2C && !HAS_DTS_I2C_DEVICE) || LSM6DSL_SPI```

Reply:
:(